bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/ghost/GhostAsterism.scala

```diff
@@ -0,0 +1,368 @@
+package edu.gemini.spModel.gemini.ghost
+
+import edu.gemini.spModel.core.{Coordinates, Magnitude, MagnitudeBand, MagnitudeSystem, Offset}
+import edu.gemini.spModel.target.SPTarget
+import edu.gemini.spModel.target.env.Asterism
+
+import java.time.{Duration, Instant}
+
+import scalaz._
+import Scalaz._
+
+/** Base trait for the three GHOST asterism types: two target, beam switching,
+  * and high resolution.
+  */
+sealed trait GhostAsterism extends Asterism {
+  import GhostAsterism.{XBinning, YBinning}
+
+  def xBinning: XBinning
+  def yBinning: YBinning
+}
+
+/** GHOST asterism model.
+  */
+object GhostAsterism {
+
+  /** The GHOST guide fibers can be enabled or disabled for each science target.
+    * Typically they are enabled for bright targets (< B mag 18 for standard
+    * resolution, < B mag 17 for high resolution) but disabled for faint targets.
+    * OIWFS state can also be explicitly disabled, say, in a crowded field where
+    * OIWFS is less effective.
+    */
+  sealed trait OiwfsState extends Product with Serializable
+
+  object OiwfsState {
+    case object Enabled  extends OiwfsState
+    case object Disabled extends OiwfsState
+
+    val enabled: OiwfsState  = Enabled
+    val disabled: OiwfsState = Disabled
+
+    val All = NonEmptyList(enabled, disabled)
+
+    implicit val EqualOiwfsState = Equal.equalA[OiwfsState]
+  }
+
+
+  /** GHOST targets are associated with a guiding state (enabled or disabled),
+    * referring to whether the dedicated guide fibers surrounding the science
+    * target should be used.
+    *
```

Comment:
The "GemOps v3" document refers to these as OIWFS so I must have picked that up from there.  I'll make it more consistent though.